libsolidity/formal/SSAVariable.cpp

```diff
@@ -31,3 +31,10 @@ void SSAVariable::resetIndex()
 	m_currentIndex = 0;
 	m_nextFreeIndex = make_unique<unsigned>(1);
 }
+
+void SSAVariable::setIndex(unsigned _index)
+{
+	m_currentIndex = _index;
+	if (*m_nextFreeIndex <= _index)
```

Comment:
Without thinking about this too much, but from looking at the code in the diff: Why exactly is ``m_nextFreeIndex`` a unique pointer :-)? Unrelated to this PR, though.
But I mean: either this can be unset, then we should at least check or assert here - or it *can't* be unset, then it doesn't need to be a unique pointer, does it?

Reply:
This is for the case where SSAVariables are copied, which happens when there are branches. In that case, we want all the SSAVariables that refer to the same program variable to have the same `next` which must be updated if any of those update it.

Reply:
But - by definition - you can't share a unique pointer!

Reply:
Can you point me to the code that does this? If you have two unique_ptr's containing the same pointer, then you'll end up with double-frees...

Reply:
I mean - an SSAVariable with a unique_ptr won't be copiable, so apparently SSAVariables *aren't* copied... and the ``m_nextFreeIndex`` is private and only used with by dereferencing or by assigning ``= make_unique<unsigned>(1);``... to me it very much looks like this could just be a plain member ``unsigned m_nextFreeIndex = 1;`` :-).

Reply:
Damn, true. Then I'd rather make it a `shared_ptr` though

Reply:
Tbh then I don't know how it's working right now haha

Reply:
Yeah - either this doesn't need to be a pointer at all, or there's something fishy about this, if you think it has to be :-D.

Reply:
Removed pointer, let's see